service/storage_service.cc

```diff
@@ -3004,10 +3004,8 @@ future<> storage_service::shutdown_protocol_servers() {
 }
 
 std::unordered_multimap<inet_address, dht::token_range>
-storage_service::get_new_source_ranges(const sstring& keyspace_name, const dht::token_range_vector& ranges) const {
+storage_service::get_new_source_ranges(locator::effective_replication_map_ptr erm, const dht::token_range_vector& ranges) const {
     auto my_address = get_broadcast_address();
-    auto& ks = _db.local().find_keyspace(keyspace_name);
-    auto erm = ks.get_effective_replication_map();
     std::unordered_map<dht::token_range, inet_address_vector_replica_set> range_addresses = erm->get_range_addresses();
     std::unordered_multimap<inet_address, dht::token_range> source_ranges;
 
```

Comment:
Ok, I see it calls _gossiper.is_alive(), so these mix run-time information with the inputs. So can't change here.